java-frontend/src/main/java/org/sonar/plugins/java/api/GeneratedCodeVisitor.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.java.api;
+
+import com.google.common.annotations.Beta;
+
+/**
+ * Marker interface for code visitors which are invoked on generated code
+ */
+@Beta
+public interface GeneratedCodeVisitor extends JavaCheck {
```

Comment:
I don't see any mechanism which would prevent people writing custom rules to use this as well, expecting their code to be executed on "generated code". (code annotated with `@Generated`?).
While I understand that we don't necessarily want to expose everything, I would appreciate to have it a bit more explicit. note any kind of generated code is going to be analyzed by such rule.

Reply:
I am not sure I understand the comment, do you just want to improve javadoc?